2/chapitre_02.md

```diff
@@ -0,0 +1,197 @@
+# Résumé
+
+Le but du chapitre est de se familiariser avec le pseudo-code utilisé dans le livre et avec les
+notations liées à la complexité algorithmique. Les deux principaux exemples sont :
+
+- le tri par insertion
+- le tri fusion
+
+## Tri par insertion
+
+Note à propos du pseudo-code: le but est d'être précis tout en s'abstrayant des détails quand cela
+permet de mieux comprendre l'algorithme en lui-même. Par exemple, on peut remplacer certaines
+instructions par une phrase en langue naturelle.
+
+Le code peut être trouvé à peu près n'importe où. Il consiste à chaque tour de boucle à
+prendre le prochain élément non trié, à faire glisser tous les éléments qui lui sont supérieurs
+vers la fin du tableau et finalement l'insérer juste après le premier élément inférieur.
+
+### La notion d'invariant de boucle
+
+Un invariant de boucle est une propriété qui est vraie avant de commencer chaque itération de boucle
+(comprenant l'invalidation de la condition de boucle). Trois propriété à montrer:
```

Comment:
Il me semble que tu devrais seulement dire qu'un invariant a les trois propriétés (avec un s d'ailleurs) que tu mentionnes.

Reply:
J'essayais de simplifier l'introduction de la notion avec une première étape simple et la seconde qui dit ce qui est nécessaire. Mais la phrase n'est pas terrible quand même. Quelque chose du genre :

Un invariant est une propriété vraie avant et après chaque itération de boucle, on doit donc montrer :